src/ptvsd/_vendored/pydevd/pydevd.py

```diff
@@ -166,7 +166,10 @@ def _on_run(self):
                 self._py_db_command_thread_event.clear()
                 self._py_db_command_thread_event.wait(0.3)
         except:
-            pydev_log.debug(sys.exc_info()[0])
+            try:
+                pydev_log.debug(sys.exc_info()[0])
+            except:
+                pass  # In interpreter shutdown sys may be None already.
```

Comment:
This comment can be turned into an assert to ensure that there aren't any random exceptions getting swallowed here unexpectedly.

Reply:
Rectified the comment to:
```
# In interpreter shutdown many things can go wrong (any module variables may
# be None, streams can be closed, etc).
```

It's not really just `sys` which may be `None`, many other things could go haywire at that point. 

Note that I agree this should be handled better, but we should first have a way to identify that we're indeed in interpreter shutdown (created https://github.com/Microsoft/ptvsd/issues/1175 to address that as it's really out of the scope of this pull request -- this was mostly because of a test failure at interpreter shutdown).